Rules/CommonScripts/MapVotesCommon.as

```diff
@@ -0,0 +1,610 @@
+		
+		button3.RefreshButton( MenuSize.x, ButtonSize);
+		MenuSize.x += ButtonSize.x+30;
+		MenuSize.y = (ButtonSize.y+100) > MenuSize.y ? ButtonSize.y+95 : MenuSize.y;
+	}
+
+	void Update(CControls@ controls, u8 &out SelectedNum)
+	{
+		Vec2f mousepos = controls.getMouseScreenPos();
+		const bool mousePressed = controls.isKeyPressed(KEY_LBUTTON);
+		const bool mouseJustReleased = controls.isKeyJustReleased(KEY_LBUTTON);
+
+		int hoverednum;
+		if (button1.isHovered(mousepos))
+		{
+			if (button1.State == 3) return; 
+			else button1.State = 1; 
+			
+			if (mousePressed) button1.State = 2;
+			else if(mouseJustReleased) 
+			{
+				SelectedNum = 1;
+				button1.State = 3;
+				button2.State = button3.State = 0;
+			}
+		}
+		else if (button2.isHovered(mousepos))
+		{
+			if (button2.State == 3) return; 
+			else button2.State = 1; 
+			
+			if (mousePressed) button2.State = 2;
+			else if(mouseJustReleased) 
+			{
+				SelectedNum = 2;
+				button2.State = 3;
+				button1.State = button3.State = 0;
+			}
+		}
+		else if (button3.isHovered(mousepos))
+		{
+			if (button3.State == 3) return; 
+			else button3.State = 1; 
+			
+			if (mousePressed) button3.State = 2;
+			else if(mouseJustReleased) 
+			{
+				SelectedNum = 3;
+				button3.State = 3;
+				button1.State = button2.State = 0;
+			}
+		}
+		else
+		{
+			SelectedNum = 0;
+			button1.State = button1.State != 3 ? 0 : 3; 
+			button2.State = button2.State != 3 ? 0 : 3;
+			button3.State = button3.State != 3 ? 0 : 3;
+		}		
+	}
+
+	void RenderGUI()
+	{
+		Vec2f ScreenDim = getDriver().getScreenDimensions();
+
+		if (fadeTimer > 0 && fadeTimer < FadeTicks)
+		{
+			SColor col(colors::menu_invisible_color);
+			float fadeamount = 1.0 - (fadeTimer*0.01f);
+			col = col.getInterpolated(colors::menu_fadeout_color, fadeamount);
+
+			GUI::DrawRectangle(Vec2f_zero, ScreenDim, col);
+		}
+		else if (fadeTimer >= FadeTicks) // draw menu
+		{			
+			GUI::SetFont("menu");	
+			GUI::DrawRectangle(Vec2f_zero, ScreenDim, colors::menu_fadeout_color);
+			GUI::DrawFramedPane(TL_Position, BR_Pos);		
+
+			if (VoteTimeLeft < 1)
+			{	
+				string winner = "";
+				switch (MostVoted)
+				{
+					case 1: winner = button1.shortname; break;
+					case 3:	winner = button3.shortname; break;
+					default: winner = "A Random Map"; break;
+				}
+			 	GUI::DrawText("Map Vote Has Ended.. Loading : "+winner, TL_Position+Vec2f(20,8), color_white);
+			}
+			else
+			{
+			 	GUI::DrawText("Map Vote Ends In : "+ VoteTimeLeft +" Secs", TL_Position+Vec2f(20,8), color_white);
+			}
+			
+
+			const Vec2f NameMid1 = button1.Pos+Vec2f((button1.Size.x/2)-2, button1.Size.y + 36);
+			const Vec2f NameMid2 = button2.Pos+Vec2f((button2.Size.x/2)-2, button2.Size.y + 36);
+			const Vec2f NameMid3 = button3.Pos+Vec2f((button3.Size.x/2)-2, button3.Size.y + 36);
+			
+			GUI::SetFont("arial_20");
+			GUI::DrawTextCentered(""+ VotedCount1, NameMid1, MostVoted == 1 ? SColor(colors::green_color) : color_white);
```

Comment:
Let's not change the color of the text with the most votes. It might confuse players if that's not the one they voted for.